neural_lam/datastore/npyfilesmeps/create_parameter_weights.py

```diff
@@ -200,12 +197,13 @@ def main():
             w_list.astype("float32"),
         )
 
-    # Load dataset without any subsampling
+    # XXX: is this correct?
+    ar_steps = 61
     ds = WeatherDataset(
-        config_loader.dataset.name,
+        datastore=datastore,
         split="train",
-        subsample_step=1,
-        pred_length=63,
+        ar_steps=ar_steps,
+        # pred_length=63,
         standardize=False,
     )
```

Comment:
ah, this is maybe where I am going wrong in https://github.com/leifdenby/neural-lam/blob/feat/datastores/neural_lam/datastore/npyfilesmeps/compute_standardization_stats.py#L170. Not sure what the correct way of thinking about this should be. @sadamov this relates to the forcing window applied right?

Reply:
I can rename this file back btw if you prefer. I just thought it was clearer not having "weights" in the name since the actual weights are derived by these statistics **and** weightings provided by other means (e.g. pressure/altitude based or uniform for example)

Reply:
Maybe my tests from https://github.com/mllam/neural-lam/commit/8cc6c3dc2ff9f099f2e6ae6fcb0d77637d6b3a47 could help with understanding, as examples of how the forcing windowing affects things.


Yes this file doesn't actually compute the variable weights anymore, so I think the name change is good.

Reply:
Ah yes, this is related to the forcing and how `num_past_forcing_steps` can reuse `init_steps` (i.e. they overlap). I will take a look at the forcings again:
- Rename them
- Make sure the MEPS datastore is properly initialized
- Double check the respective tests.
Give me a day here, and it shall be fixed @joeloskarsson @leifdenby

Reply:
> For this to be correct I don't think we should do any windowing of the forcing, i.e. instantiate the dataset class with `include_past_forcing=0, include_future_forcing=0`. Then `ar_steps=61` should be correct.

@joeloskarsson Why would you prefer the above to `include_past_forcing=1, include_future_forcing=1`, `ar_steps=63`. Wouldn't the latter be closer to the original paper?

Reply:
Currently in the code it says `    ar_steps = datastore._num_timesteps - 10` not quite sure where this is coming from?

Reply:
Yes, you are correct, it should 63 :sweat_smile: Original sample length is 65, -2 init steps = 63.

No idea about the -10 though.

Reply:
Sorry, the `-10` was my fault 😔 I got confused about the windowing and couldn't get the code to execute without running beyond the length of the array (in time). I can see you fixed this in https://github.com/mllam/neural-lam/pull/66/commits/46f161c7d166bbce49933f50ec23ee237c8e41f5 @sadamov, thank you! I think we can resolve this then :)

Reply:
ah I see, clearly my bad, I struggled with time in general :watch: and had a succesion of bugs. I think, it's finally fixed indeed, will resolve here.